miniapps/meshing/mesh-optimizer.hpp

```diff
@@ -279,3 +386,959 @@ void DiffuseField(ParGridFunction &field, int smooth_steps)
    delete Lap;
 }
 #endif
+
+
+/// This class is used to update GridFunction and FESpaces after mesh updates.
+/// It also provides functionality to reset essential boundary conditions, and
+/// rebalance ParMesh.
+class TMOPAMR
```

Comment:
All these classes should be moved to `tmop_amr.hpp/cpp` in `/fem` s.t. they are available with the library.